boa/src/builtins/object/mod.rs

```diff
@@ -47,322 +46,85 @@ pub static PROTOTYPE: &str = "prototype";
-            } else {
-                self.properties.insert(property_key.clone(), current);
-            }
-        // 7
-        } else if current.is_data_descriptor() && desc.is_data_descriptor() {
-            // a
-            if !current.configurable.unwrap() && !current.writable.unwrap() {
-                if desc.writable.is_some() && desc.writable.unwrap() {
-                    return false;
-                }
-
-                if desc.value.is_some()
-                    && !same_value(
-                        &desc.value.clone().unwrap(),
-                        &current.value.clone().unwrap(),
-                        false,
-                    )
-                {
-                    return false;
-                }
-
-                return true;
-            }
-        // 8
-        } else {
-            if !current.configurable.unwrap() {
-                if desc.set.is_some()
-                    && !same_value(
-                        &desc.set.clone().unwrap(),
-                        &current.set.clone().unwrap(),
-                        false,
-                    )
-                {
-                    return false;
-                }
-
-                if desc.get.is_some()
-                    && !same_value(
-                        &desc.get.clone().unwrap(),
-                        &current.get.clone().unwrap(),
-                        false,
-                    )
-                {
-                    return false;
-                }
+impl Display for ObjectData {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result<(), Error> {
+        write!(
+            f,
+            "{}",
+            match self {
+                Self::Function(_) => "Function",
+                Self::Array => "Array",
+                Self::String(_) => "String",
+                Self::Symbol(_) => "Symbol",
+                Self::Error => "Error",
+                Self::Ordinary => "Ordinary",
+                Self::Boolean(_) => "Boolean",
+                Self::Number(_) => "Number",
+                Self::BigInt(_) => "BigInt",
             }
-
-            return true;
-        }
-        // 9
-        self.properties.insert(property_key, desc);
-        true
+        )
     }
 }
 
-impl Object {
+impl Default for Object {
     /// Return a new ObjectData struct, with `kind` set to Ordinary
-    pub fn default() -> Self {
+    fn default() -> Self {
         let mut object = Self {
-            kind: ObjectKind::Ordinary,
+            data: ObjectData::Ordinary,
             internal_slots: FxHashMap::default(),
             properties: FxHashMap::default(),
-            sym_properties: FxHashMap::default(),
+            symbol_properties: FxHashMap::default(),
             state: None,
-            func: None,
         };
 
         object.set_internal_slot("extensible", Value::from(true));
         object
     }
+}
+
+impl Object {
+    pub fn new() -> Self {
+        Default::default()
+    }
```

Comment:
I wouldn't add a `new()` function that is just a `default()`.

Reply:
This was taken from the `Rust API guideline` in `Predictability` section `Constructors are static, inherent methods (C-CTOR)` 

> Note that it is common and expected for types to implement both Default and a new constructor. For types that have both, they should have the same behavior. Either one may be implemented in terms of the other.

I think eventually will have to remove them, there should because it does not create an object with the prototype, also the rule only applies if its a part of the api.

Reply:
Yeah i think this should just implement `Default` and we move to using that

Reply:
I don't know, `clippy` seems to recommend not having a `new()` function without arguents, and using a `default()` instead. In any case, I think we should mark it as `#[inline]`.